backend/utils/authutils/tvault.go

```diff
@@ -0,0 +1,89 @@
+package authutils
+
+import (
+	"Tella-Desktop/backend/utils/constants"
+	"io"
+	"os"
+)
+
+// TVault helper methods -> move this to utils
+func WriteTVaultHeader(salt, encryptDBKey []byte) error {
+	//Create tvault file if it doesn't exist
+	file, err := os.Create(GetTVaultPath())
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Write version
+	if _, err := file.Write([]byte{1}); err != nil {
+		return err
+	}
+
+	// write salt
+	if _, err := file.Write(salt); err != nil {
+		return err
+	}
+
+	// write encrypted db key
+	if _, err := file.Write(encryptDBKey); err != nil {
+		return err
+	}
+
+	headerSize := 1 + len(salt) + len(encryptDBKey)
+	if headerSize < constants.TVaultHeaderSize {
+		padding := make([]byte, constants.TVaultHeaderSize-headerSize)
+		if _, err := file.Write(padding); err != nil {
+			return err
+		}
```

Comment:
in your `file.Write` calls, the first return value is the number of bytes written. i think using the ints returned from that call to build up var `headerSize` is better because then you have something that is actually representative of what's on disk; could be that the num bytes written is different than what you think in your arithmetic due to differences when encoding in some step